### PR TITLE
Make color immutable

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Makes `Color` immutable. This more closely aligns with the Java Color class and makes behavior consistent between Playground and Theater.

JIRA: https://codedotorg.atlassian.net/browse/CSA-865

Testing: local & unit

### CODE REVIEW

File: org-code-javabuilder/media/src/main/java/org/code/media/Color.java [hunk 1]
```diff
@@ -24,7 +36,7 @@ public Color(String color) throws IllegalArgumentException {
     this.red = colorToCopy.getRed();
     this.green = colorToCopy.getGreen();
     this.blue = colorToCopy.getBlue();
-    this.setFullOpacity();
+    this.alpha = 255;
```
Comment (molly-moen) [inline org-code-javabuilder/media/src/main/java/org/code/media/Color.java:39]: nit: create constant for 255

File: org-code-javabuilder/media/src/test/java/org/code/media/ColorTest.java [hunk 2]
```diff
@@ -12,11 +12,38 @@ public void defaultAlphaIs255() {
...
-    assertEquals(255, c.getAlpha());
+  public void testCopyWithRedCreatesCopyWithNewRedValue() {
+    final int red = 100;
+    final int green = 150;
+    final int blue = 200;
+    final int newRed = 50;
+    final Color copied = Color.copyWithRed(new Color(red, green, blue), newRed);
+    assertEquals(newRed, copied.getRed());
+    assertEquals(green, copied.getGreen());
+    assertEquals(blue, copied.getBlue());
```
Comment (molly-moen) [inline org-code-javabuilder/media/src/test/java/org/code/media/ColorTest.java:23]: can you also add a check that `copied` does not equal (strict equality) the original color?

**Review by molly-moen — APPROVED** [review]
LGTM, just a couple nits

File: org-code-javabuilder/media/src/test/java/org/code/media/ColorTest.java (hunk 2 above)
Comment (sanchitmalhotra126) [reply org-code-javabuilder/media/src/test/java/org/code/media/ColorTest.java:26]: Sure, done!

**Pull request merged**